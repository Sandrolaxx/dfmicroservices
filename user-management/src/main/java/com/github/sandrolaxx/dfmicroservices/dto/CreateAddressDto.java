package com.github.sandrolaxx.dfmicroservices.dto;

public class CreateAddressDto {

  public String state;

  public String city;

  public String district;

  public String street;

  public Integer number;

  public Integer numberAp;

  public Double latitude;

  public Double longitude;

  public boolean active;

  public boolean main;

  public String getState() {
    return this.state;
  }

  public void setState(String state) {
    this.state = state;
  }

  public String getCity() {
    return this.city;
  }

  public void setCity(String city) {
    this.city = city;
  }

  public String getDistrict() {
    return this.district;
  }

  public void setDistrict(String district) {
    this.district = district;
  }

  public String getStreet() {
    return this.street;
  }

  public void setStreet(String street) {
    this.street = street;
  }

  public Integer getNumber() {
    return this.number;
  }

  public void setNumber(Integer number) {
    this.number = number;
  }

  public Integer getNumberAp() {
    return this.numberAp;
  }

  public void setNumberAp(Integer numberAp) {
    this.numberAp = numberAp;
  }

  public Double getLatitude() {
    return this.latitude;
  }

  public void setLatitude(Double latitude) {
    this.latitude = latitude;
  }

  public Double getLongitude() {
    return this.longitude;
  }

  public void setLongitude(Double longitude) {
    this.longitude = longitude;
  }

  public boolean isActive() {
    return this.active;
  }

  public void setActive(boolean active) {
    this.active = active;
  }

  public boolean isMain() {
    return this.main;
  }

  public void setMain(boolean main) {
    this.main = main;
  }

}
