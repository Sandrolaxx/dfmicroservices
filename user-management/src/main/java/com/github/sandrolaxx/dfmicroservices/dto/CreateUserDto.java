package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

public class CreateUserDto {

  public String name;

  public String email;

  public String password;

  public String document;

  public String phone;

  public boolean active;

  public List<CreateAddressDto> address;

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getEmail() {
    return this.email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public String getPassword() {
    return this.password;
  }

  public void setPassword(String password) {
    this.password = password;
  }

  public String getDocument() {
    return this.document;
  }

  public void setDocument(String document) {
    this.document = document;
  }

  public String getPhone() {
    return this.phone;
  }

  public void setPhone(String phone) {
    this.phone = phone;
  }

  public boolean getActive() {
    return this.active;
  }

  public void setActive(boolean active) {
    this.active = active;
  }

  public List<CreateAddressDto> getAddress() {
    return this.address;
  }

  public void setAddress(List<CreateAddressDto> address) {
    this.address = address;
  }
  
}
