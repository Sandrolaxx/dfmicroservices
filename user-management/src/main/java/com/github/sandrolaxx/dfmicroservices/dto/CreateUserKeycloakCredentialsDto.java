package com.github.sandrolaxx.dfmicroservices.dto;

public class CreateUserKeycloakCredentialsDto {

    private String value;

    private boolean temporary;

    public String getValue() {
        return this.value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public boolean getTemporary() {
        return this.temporary;
    }

    public void setTemporary(boolean temporary) {
        this.temporary = temporary;
    }

}