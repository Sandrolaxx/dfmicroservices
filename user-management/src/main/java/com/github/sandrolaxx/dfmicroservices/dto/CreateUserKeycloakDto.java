package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

public class CreateUserKeycloakDto {

    private String username;

    private List<CreateUserKeycloakCredentialsDto> credentials;

    private boolean enabled;

    public String getUsername() {
        return this.username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public List<CreateUserKeycloakCredentialsDto> getCredentials() {
        return this.credentials;
    }

    public void setCredentials(List<CreateUserKeycloakCredentialsDto> credentials) {
        this.credentials = credentials;
    }

    public boolean isEnabled() {
        return this.enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

}