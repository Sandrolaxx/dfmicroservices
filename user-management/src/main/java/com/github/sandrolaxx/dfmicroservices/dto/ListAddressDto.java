package com.github.sandrolaxx.dfmicroservices.dto;

public class ListAddressDto {

    public Integer id;

    public String state;

    public String city;

    public String district;

    public String street;

    public Integer number;

    public Integer numberAp;

    public Double latitude;

    public Double longitude;

    public boolean main;

    public Integer getId() {
        return this.id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getState() {
        return this.state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getCity() {
        return this.city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getDistrict() {
        return this.district;
    }

    public void setDistrict(String district) {
        this.district = district;
    }

    public String getStreet() {
        return this.street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public Integer getNumber() {
        return this.number;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }

    public Integer getNumberAp() {
        return this.numberAp;
    }

    public void setNumberAp(Integer numberAp) {
        this.numberAp = numberAp;
    }

    public Double getLatitude() {
        return this.latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return this.longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public boolean getMain() {
        return this.main;
    }

    public void setMain(boolean main) {
        this.main = main;
    }

}
