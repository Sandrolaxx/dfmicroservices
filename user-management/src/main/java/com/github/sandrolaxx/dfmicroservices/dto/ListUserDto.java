package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

public class ListUserDto {

  public Integer id;

  public String name;

  public String email;

  public String password;

  public String document;

  public String phone;

  public boolean active;

  public List<ListAddressDto> address;

  public String createdAt;

  public String updatedAt;

  public Integer getId() {
    return this.id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getEmail() {
    return this.email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public String getPassword() {
    return this.password;
  }

  public void setPassword(String password) {
    this.password = password;
  }

  public String getDocument() {
    return this.document;
  }

  public void setDocument(String document) {
    this.document = document;
  }

  public String getPhone() {
    return this.phone;
  }

  public void setPhone(String phone) {
    this.phone = phone;
  }

  public boolean getActive() {
    return this.active;
  }

  public void setActive(boolean active) {
    this.active = active;
  }

  public List<ListAddressDto> getAddress() {
    return this.address;
  }

  public void setAddress(List<ListAddressDto> address) {
    this.address = address;
  }

  public String getCreatedAt() {
    return this.createdAt;
  }

  public void setCreatedAt(String createdAt) {
    this.createdAt = createdAt;
  }

  public String getUpdatedAt() {
    return this.updatedAt;
  }

  public void setUpdatedAt(String updatedAt) {
    this.updatedAt = updatedAt;
  }

}
