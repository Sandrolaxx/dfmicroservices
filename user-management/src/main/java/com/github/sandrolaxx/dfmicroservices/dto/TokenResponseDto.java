package com.github.sandrolaxx.dfmicroservices.dto;

import javax.json.bind.annotation.JsonbProperty;

public class TokenResponseDto {

    @JsonbProperty("access_token")
    private String accessToken;

    @JsonbProperty("expires_in")
    private Integer expiresIn;

    @JsonbProperty("refresh_expires_in")
    private Integer refreshExpiresIn;

    @JsonbProperty("refresh_token")
    private String refreshToken;

    @JsonbProperty("token_type")
    private String tokenType;

    @JsonbProperty("session_state")
    private String sessionState;

    @JsonbProperty("scope")
    private String scope;

    public String getAccessToken() {
        return this.accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public Integer getExpiresIn() {
        return this.expiresIn;
    }

    public void setExpiresIn(Integer expiresIn) {
        this.expiresIn = expiresIn;
    }

    public Integer getRefreshExpiresIn() {
        return this.refreshExpiresIn;
    }

    public void setRefreshExpiresIn(Integer refreshExpiresIn) {
        this.refreshExpiresIn = refreshExpiresIn;
    }

    public String getRefreshToken() {
        return this.refreshToken;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public String getTokenType() {
        return this.tokenType;
    }

    public void setTokenType(String tokenType) {
        this.tokenType = tokenType;
    }

    public String getSessionState() {
        return this.sessionState;
    }

    public void setSessionState(String sessionState) {
        this.sessionState = sessionState;
    }

    public String getScope() {
        return this.scope;
    }

    public void setScope(String scope) {
        this.scope = scope;
    }

}
