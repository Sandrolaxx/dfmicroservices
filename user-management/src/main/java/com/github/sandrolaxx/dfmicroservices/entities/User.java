package com.github.sandrolaxx.dfmicroservices.entities;

import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.PrePersist;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Transient;

import com.github.sandrolaxx.dfmicroservices.entities.enums.EnumMessageType;
import com.github.sandrolaxx.dfmicroservices.utils.EncryptUtil;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.jboss.aerogear.security.otp.api.Base32;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

@Entity
@Table(name = "DF_USER")
public class User extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "NAME")
    private String name;

    @Column(name = "EMAIL")
    private String email;

    @Column(name = "PASSWORD")
    private String password;

    @Column(name = "DOCUMENT")
    private String document;

    @Column(name = "PHONE")
    private String phone;

    @Column(name = "ACTIVE")
    private boolean active;

    @Column(name = "ACCEPT_TERMS")
    private boolean acceptTerms;

    @Column(name = "SECRET")
    private String secret;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_USER")
    private List<Address> address;

    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "CREATED_AT")
    private Date createdAt;

    @UpdateTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "UPDATED_AT")
    private Date updatedAt;

    @Transient
    private EnumMessageType messageType;

    @PrePersist
    private void encryptSensitiveData() {
        this.document = EncryptUtil.textEncrypt(this.document, secret.substring(0, 16));
        this.email = EncryptUtil.textEncrypt(this.email, secret.substring(0, 16));
        this.password = EncryptUtil.textEncrypt(this.password, secret.substring(0, 16));
        this.phone = EncryptUtil.textEncrypt(this.phone, secret.substring(0, 16));
        this.name = EncryptUtil.textEncrypt(this.name, secret.substring(0, 16));
    }

    public User() {
        super();
        this.secret = Base32.random();
    }

    public Integer getId() {
        return this.id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return this.password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getDocument() {
        return this.document;
    }

    public void setDocument(String document) {
        this.document = document;
    }

    public String getPhone() {
        return this.phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getSecret() {
        return this.secret;
    }

    public void setSecret(String secret) {
        this.secret = secret;
    }

    public boolean isActive() {
        return this.active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public boolean isAcceptTerms() {
        return this.acceptTerms;
    }

    public void setAcceptTerms(boolean acceptTerms) {
        this.acceptTerms = acceptTerms;
    }

    public List<Address> getAddress() {
        return this.address;
    }

    public void setAddress(List<Address> address) {
        this.address = address;
    }

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return this.updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public EnumMessageType getMessageType() {
        return this.messageType;
    }

    public void setMessageType(EnumMessageType messageType) {
        this.messageType = messageType;
    }

}
