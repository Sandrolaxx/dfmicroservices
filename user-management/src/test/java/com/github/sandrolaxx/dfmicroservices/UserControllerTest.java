package com.github.sandrolaxx.dfmicroservices;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import javax.ws.rs.core.Response.Status;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.sandrolaxx.dfmicroservices.dto.CreateAddressDto;
import com.github.sandrolaxx.dfmicroservices.dto.CreateUserDto;
import com.github.sandrolaxx.dfmicroservices.entities.User;
import com.github.sandrolaxx.dfmicroservices.util.TokenUtils;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(UserTestLifecycleManager.class)
public class UserControllerTest {

    private String token;

    @BeforeEach
    public void genereteToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTDonaFrostClaims.json", null);
    }
    
    private RequestSpecification given() {
        return RestAssured.given()
        .contentType(ContentType.JSON).header(new Header("Authorization", "Bearer " + token));
    }

    private RequestSpecification givenNoToken() {
        return RestAssured.given()
        .contentType(ContentType.JSON);
    }

    @Test
    @DataSet("user-cenario1.json")
    public void testFindUsers() {
        String result = given()
            .when().get("/dona-frost/v1/user")
            .then()
            .statusCode(200)
            .extract().asString();

            Approvals.verifyJson(result);
    }

    @Test
    @DataSet("user-cenario1.json")
    public void testFindUsersNoToken() {
        givenNoToken()
            .when().get("/dona-frost/v1/user")
            .then()
            .statusCode(401)
            .extract().asString();
    }

    @Test
    @DataSet(value = "user-cenario1.json")
    public void testCreateUser() {
        CreateUserDto dto = new CreateUserDto();
        List<CreateAddressDto> addressList = new ArrayList<>();
        CreateAddressDto addressDto = new CreateAddressDto();

        addressDto.setActive(true);
        addressDto.setState("PR");
        addressDto.setCity("Londrina");
        addressDto.setDistrict("Bairro jardim XX");
        addressDto.setStreet("Rua elias abraão");
        addressDto.setNumber(91);
        addressDto.setNumberAp(63);
        addressDto.setMain(true);
        addressDto.setLatitude(4343443D);
        addressDto.setLongitude(434343D);

        addressList.add(addressDto);

        dto.setAddress(addressList);
        dto.setName("Roberto");
        dto.setDocument("10564574902");
        dto.setEmail("teste@hotmail.com");
        dto.setPassword("12222");
        dto.setPhone("12564588885");;
        dto.setActive(true);


        String result = given()
            .body(dto)
            .when().post("/dona-frost/v1/user")
            .then()
            .statusCode(201)
            .extract().asString();

        //Somente aplicação rodando, não é possível criar transação no test    
        Approvals.verifyJson(result);
    }


    @Test
    @DataSet("user-cenario1.json")
    public void testChangeUser() {
        CreateUserDto dto = new CreateUserDto();

        dto.setName("Roberto");
        dto.setDocument("10564574902");
        dto.setEmail("teste@hotmail.com");
        dto.setPassword("12222");
        dto.setPhone("12564588885");;
        dto.setActive(true);

        Integer idUserToChange = 1;

        given()
            .with().pathParam("id", idUserToChange)
            .body(dto)
            .when().put("/dona-frost/v1/user/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode())
            .extract().asString();

        User findById = User.findById(idUserToChange);
        
        Assert.assertEquals(dto.name, findById.name);
        Assert.assertEquals(dto.document, findById.document);
        Assert.assertEquals(dto.email, findById.email);
        Assert.assertEquals(dto.phone, findById.phone);
        Assert.assertEquals(dto.password, findById.password);
    }

    @Test
    @DataSet("user-cenario1.json")
    public void testDeleteUser() {

        Integer idUserToDelete = 1;

        given()
            .with().pathParam("id", idUserToDelete)
            .when().delete("/dona-frost/v1/user/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode())
            .extract().asString();

        User findById = User.findById(idUserToDelete);
        
        Assert.assertEquals(findById, null);

    }

}