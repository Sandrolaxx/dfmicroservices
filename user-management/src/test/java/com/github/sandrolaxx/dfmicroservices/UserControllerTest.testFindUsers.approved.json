[
  {
    "active": true,
    "address": [
      {
        "active": true,
        "city": "Cascavel",
        "district": "Interlagos",
        "id": 1,
        "latitude": 4343434.0,
        "longitude": 4545454.0,
        "main": true,
        "number": 33,
        "state": "PR",
        "street": "Rua grevilha"
      }
    ],
    "createdAt": "03/04/2021 14:48:42",
    "document": "123144333434",
    "email": "teste1@hotmail.com",
    "id": 1,
    "name": "Cleber da Siltva",
    "password": "123232",
    "phone": "45991033232",
    "updatedAt": "03/04/2021 14:48:42"
  },
  {
    "active": true,
    "address": [
      {
        "active": true,
        "city": "Cascavel",
        "district": "Centro",
        "id": 2,
        "latitude": 4343434.0,
        "longitude": 4545454.0,
        "main": true,
        "number": 22,
        "numberAp": 78,
        "state": "PR",
        "street": "Blue baron"
      }
    ],
    "createdAt": "03/04/2021 14:48:42",
    "document": "10564574902",
    "email": "teste2@hotmail.com",
    "id": 2,
    "name": "gabrielli vieirra",
    "password": "saporoxo159753",
    "phone": "4598147893",
    "updatedAt": "03/04/2021 14:48:42"
  },
  {
    "active": true,
    "address": [
      {
        "active": true,
        "city": "Cascavel",
        "district": "Interlagos",
        "id": 3,
        "latitude": 4343434.0,
        "longitude": 4545454.0,
        "main": true,
        "number": 33,
        "state": "PR",
        "street": "Rua grevilha"
      }
    ],
    "createdAt": "03/04/2021 14:48:42",
    "document": "10563333332",
    "email": "sandro@hotmail.com",
    "id": 3,
    "name": "sandro matheus ramos",
    "password": "1329",
    "phone": "45991039812",
    "updatedAt": "03/04/2021 14:48:42"
  }
]